build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/tscTask.ts

```diff
@@ -87,7 +87,10 @@ export class TscTask extends LeafTask {
 				if (this._projectReference) {
 					fullPath = this._projectReference.remapSrcDeclFile(fullPath, config);
 				}
-				const hash = await this.node.buildContext.fileHashCache.getFileHash(fullPath);
+				const hash = await this.node.buildContext.fileHashCache.getFileHash(
+					fullPath,
+					TscUtils.getSourceFileVersion,
+				);
```

Comment:
...and here is where we substitute TypeScript's hashing function instead of vanilla sha256.